crates/sui-types/src/crypto.rs

```diff
@@ -239,14 +242,32 @@ pub enum PublicKey {
     Ed25519(Ed25519PublicKeyAsBytes),
     Secp256k1(Secp256k1PublicKeyAsBytes),
     Secp256r1(Secp256r1PublicKeyAsBytes),
+    ZkLogin(ZkLoginPublicIdentifier),
 }
 
+/// A wrapper struct to retrofit in [enum PublicKey] for zkLogin.
+/// Useful to construct [struct MultiSigPublicKey].
+#[derive(Clone, Debug, PartialEq, Eq, JsonSchema, Serialize, Deserialize)]
+pub struct ZkLoginPublicIdentifier(pub Vec<u8>);
+
+impl ZkLoginPublicIdentifier {
+    /// Consists of iss_bytes_len || iss_bytes || address_seed.
+    pub fn new(iss: &str, address_seed: &str) -> SuiResult<Self> {
+        let mut bytes = Vec::new();
+        let iss_bytes = iss.as_bytes();
+        bytes.extend([iss_bytes.len() as u8]);
+        bytes.extend(iss_bytes);
+        bytes.extend(big_int_str_to_bytes(address_seed).map_err(|_| SuiError::InvalidAddress)?);
+        Ok(Self(bytes))
+    }
+}
 impl AsRef<[u8]> for PublicKey {
     fn as_ref(&self) -> &[u8] {
         match self {
             PublicKey::Ed25519(pk) => &pk.0,
             PublicKey::Secp256k1(pk) => &pk.0,
             PublicKey::Secp256r1(pk) => &pk.0,
+            PublicKey::ZkLogin(z) => &z.0,
```

Comment:
change to  PublicKey::ZkLogin(z) => &bcs::to_byte(z)

Reply:
we do not derive sui address with the encoded length, so we do not do that here either.